### PR TITLE
fix hardcoded urls

### PR DESCRIPTION
**Pull request by mb-wali**

### CODE REVIEW

File: src/components/dashboard/dashboardItem/Banner.js [hunk 1]
```diff
@@ -30,6 +31,9 @@ export default function Banner(props) {
     const router = useRouter();
     const [config] = useConfig();
 
+    const { publicRuntimeConfig = {} } = getConfig() || {};
+    const cyverse_url = publicRuntimeConfig.CYVERSE_URL;
```
Comment (psarando) [inline src/components/dashboard/dashboardItem/Banner.js:35]:
```suggestion
    const cyverse_url = config?.cyverseUrl;
```

File: src/components/help/HelpTopics.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ import constants from "../../constants";
 import ProductTour from "components/help/ProductTour";
 import Feedback from "components/help/Feedback";
 import { intercomShow } from "common/intercom";
+import getConfig from "next/config";
```
Comment (psarando) [inline src/components/help/HelpTopics.js:15]:
```suggestion
import { useConfig } from "contexts/config";
```

File: src/components/help/HelpTopics.js [hunk 3]
```diff
@@ -81,6 +82,11 @@ export default function HelpTopics(props) {
     const [feedbackOpen, setFeedbackOpen] = useState();
     const classes = useStyles();
     const { t } = useTranslation("help");
+
+    const { publicRuntimeConfig = {} } = getConfig() || {};
+    const support_email = publicRuntimeConfig.SUPPORT_EMAIL;
+    const de_faq = publicRuntimeConfig.DE_FAQ;
```
Comment (psarando) [inline src/components/help/HelpTopics.js:88]:
```suggestion
    const [config] = useConfig();
    const support_email = config?.supportEmail;
    const de_faq = config?.deFaq;
```

File: src/components/layout/AppBar.js [hunk 4]
```diff
@@ -80,6 +81,9 @@ const GlobalSearchField = dynamic(() => import("../search/GlobalSearchField"));
 
 const ENTITLEMENT = "entitlement";
 
+const { publicRuntimeConfig = {} } = getConfig() || {};
+const cyverse_url = publicRuntimeConfig.CYVERSE_URL;
```
Comment (psarando) [inline src/components/layout/AppBar.js:85]:
```suggestion
const cyverse_url = config?.cyverseUrl;
```

**Review by psarando — CHANGES_REQUESTED** [review]
Looks good, except that we have a pattern of adding the next.js config values to a context, and accessing them from a `useConfig` hook in the client-side (this is actually called in a couple of the components updated in this PR).

For consistency, could you please add these config values to the context's state in `src/pages/_app.js` (similar to the way `LEGACY_DE_URL` is added):
https://github.com/cyverse-de/sonora/blob/2d22d413598db88206fb043af7f2b45b8506248e/src/pages/_app.js#L137-L246

See the `CustomIntercom` component for an example of the `useConfig` hook (but I also added some suggestions on usage in this PR):
https://github.com/cyverse-de/sonora/blob/2d22d413598db88206fb043af7f2b45b8506248e/src/components/layout/CustomIntercom.js#L25-L43

**Comment by mb-wali** [comment]
@psarando - Thank you so much for the review and helpful tips, i have push new commit with the changes you have requested. please let me know if that is in order.

**Review by psarando — APPROVED** [review]
Thank you @mb-wali, LGTM 👍

**Pull request merged**